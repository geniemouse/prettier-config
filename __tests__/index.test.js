const prettierConfig = require("../");

describe("@geniemouse/prettier-config:", () => {
    test("configuration object exists", () => {
        expect(prettierConfig).not.toBeUndefined();
    });

    test("matches the expected output", () => {
        expect(prettierConfig).toMatchSnapshot();
    });

    test("check overrides property has two entries", () => {
        expect(prettierConfig).toHaveProperty("overrides", expect.any(Array));
        expect(prettierConfig.overrides).toHaveLength(2);
    });

    describe("check boolean value properties", () => {
        test.each([
            "bracketSpacing",
            "jsxBracketSameLine",
            "jsxSingleQuote",
            "requirePragma",
            "semi",
            "singleQuote",
            "useTabs"
        ])("%s returns a boolean", (key) => {
            expect(prettierConfig).toHaveProperty(key, expect.any(Boolean));
        });
    });

    describe("check integer value properties", () => {
        test.each(["printWidth", "tabWidth"])("%s returns an integer", (key) => {
            expect(prettierConfig).toHaveProperty(key, expect.any(Number));
            expect(prettierConfig[key]).not.toBeNaN();
        });
    });

    describe("check string value properties", () => {
        test.each([
            "$schema",
            "arrowParens",
            "endOfLine",
            "htmlWhitespaceSensitivity",
            "proseWrap",
            "quoteProps",
            "trailingComma"
        ])("%s returns a string", (key) => {
            expect(prettierConfig).toHaveProperty(key, expect.any(String));
        });
    });
});
