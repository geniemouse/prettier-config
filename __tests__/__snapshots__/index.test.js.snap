// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@geniemouse/prettier-config: matches the expected output 1`] = `
Object {
  "$schema": "http://json.schemastore.org/prettierrc",
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "jsxBracketSameLine": true,
  "jsxSingleQuote": false,
  "overrides": Array [
    Object {
      "files": Array [
        ".*rc",
        "package.json",
      ],
      "options": Object {
        "parser": "json-stringify",
        "requirePragma": false,
        "tabWidth": 2,
        "trailingComma": "none",
      },
    },
    Object {
      "excludeFiles": Array [
        ".*rc",
        "package.json",
      ],
      "files": Array [
        "*.json",
      ],
      "options": Object {
        "parser": "json",
        "requirePragma": false,
        "tabWidth": 2,
        "trailingComma": "none",
      },
    },
  ],
  "printWidth": 120,
  "proseWrap": "preserve",
  "quoteProps": "consistent",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 4,
  "trailingComma": "none",
  "useTabs": false,
}
`;
